src/picongpu/include/particles/ionization/byField/BSI/BSI.hpp

```diff
@@ -54,8 +62,62 @@ namespace ionization
             /* alias for electron species to be created */
             PMACC_AUTO(electrons,pst[typename MakeIdentifier<T_DestSpecies>::type()]);
             /* call the ionization function */
-            src.ionize(electrons, currentStep);
+
+            typedef T_SrcSpecies SrcSpeciesType;
+            typedef typename SrcSpeciesType::FrameType FrameType;
+            
+            DataConnector &dc = Environment<>::get().DataConnector();
+
+            fieldE = &(dc.getData<FieldE > (FieldE::getName(), true));
+            fieldB = &(dc.getData<FieldB > (FieldB::getName(), true));
+            
+            typedef typename PMacc::traits::Resolve<
+                typename GetFlagType<FrameType,interpolation<> >::type
+            >::type InterpolationScheme;
+            
+            /* margins around the supercell for the interpolation of the field on the cells */
+            typedef typename GetMargin<InterpolationScheme>::LowerMargin LowerMargin;
+            typedef typename GetMargin<InterpolationScheme>::UpperMargin UpperMargin;
+            
+            /* define type of frame ionizer 
+             * that moves over the frames with the particles and executes an operation on them */
+            typedef particles::ionization::IonizeParticlesPerFrame<IonizationAlgorithm, 
+                MappingDesc::SuperCellSize,
+                InterpolationScheme,
+                fieldSolver::NumericalCellType > FrameIonizer;
+            
+            /* relevant area of a block */
+            typedef SuperCellDescription<
+                typename MappingDesc::SuperCellSize,
+                LowerMargin,
+                UpperMargin
+                > BlockArea;
+            
+            /* 3-dim vector : number of threads to be started in every dimension */
+            dim3 block( MappingDesc::SuperCellSize::toRT().toDim3() );
+            
+            MappingDesc cellDescription;
```

Comment:
add a constructor `BSI( MappingDesc* cellDescription) : cellDescription(cellDescription);` and a member `MappingDesc *cellDescription` to store it.